import os
import sys

# Add the project root to PYTHONPATH
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))

from custom_logging.my_logger import logger

class seleniumBotException(Exception):
    def __init__(self,error_message,error_details:sys):
        self.error_message = error_message
        _,_,exc_tb = error_details.exc_info()
        
        self.lineno=exc_tb.tb_lineno
        self.file_name=exc_tb.tb_frame.f_code.co_filename 

    def __str__(self):
        print(f"\n游댮郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷游댮\n")
        error = "Error occured in python script name [{0}] line number [{1}] error message [{2}]".format(
        self.file_name, self.lineno, str(self.error_message))
        return error
    
if __name__ == '__main__':
    try:
        a=1/0
        print("This will not be printed",a)
    except Exception as e:
        print(f"\n游댮郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷郊석둷游댮\n")
        logger.error(f"An error occurred: {str(e)}")  # Log the error first
        raise seleniumBotException(e, sys)  # Raise the custom exception
        
        